src/TumblThree/TumblThree.Applications/Downloader/FileDownloader.cs

```diff
@@ -55,7 +55,7 @@ public async Task<bool> DownloadFileWithResumeAsync(string url, string destinati
 
             if (ct.IsCancellationRequested) return false;
 
-            var fileMode = totalBytesReceived > 0 ? FileMode.Append : FileMode.Create;
+            var fileMode = File.Exists(destinationPath) ? FileMode.Append : FileMode.Create;
```

Comment:
That's another (unnecessary) filesystem access. But besides that I just see an error and an optimization possibility in the previous block (old code).

Reply:
I'm not sure the previous code is working fine for a created file of 0 bytes. Testcase may be needed.

Reply:
Sorry, I was unclear, I meant the lines 41-54 of the old code (not your changes).
I would try to avoid another filesystem access (`File.Exists`). But as said WIP...